src/Chainweb/Miner/RestAPI/Server.hs

```diff
@@ -115,15 +115,21 @@ workHandler' mr mcid m = do
     pact = _webPactExecutionService . _webBlockPayloadStorePact $ view cutDbPayloadStore cdb
 
 solvedHandler
-    :: forall l cas. Logger l => MiningCoordination l cas -> HeaderBytes -> IO NoContent
+    :: forall l cas. Logger l => MiningCoordination l cas -> HeaderBytes -> Handler NoContent
 solvedHandler mr (HeaderBytes hbytes) = do
-    ms <- readTVarIO tms
-    bh <- runGet decodeBlockHeaderWithoutHash hbytes
-    publish lf ms (_coordCutDb mr) bh
-    let !phash = _blockPayloadHash bh
-        !bct = _blockCreationTime bh
-    atomically . modifyTVar' tms . over _Unwrapped $ M.delete (T2 bct phash)
-    pure NoContent
+    ms <- liftIO $ readTVarIO tms
+    liftIO (try $ runGet decodeBlockHeaderWithoutHash hbytes) >>= \case
+        Left (DecodeException e) -> do
+            let err = TL.encodeUtf8 $ TL.fromStrict e
+            throwError err400 { errBody = "Decoding error: " <> err }
+        Left _ ->
+            throwError err400 { errBody = "General exception occured in /solved handler" }
```

Comment:
Are you sure that is does what you expect it to do? It will only catch exceptions of type `EncodingExceptions`. It won't catch `SomeException`.

Reply:
You may want to use `tryAll` to handle `SomeException` and use `toException` to pattern match on the inner exception types. Or use `catches` with `Handler`.

If you intend to only catch `EncodingException` values here, you may change the "General" error message.